vignettes/algorithm.Rmd

```diff
@@ -157,6 +157,22 @@ The parametrization of $\theta$ is similar to other non-grouped $\theta$.
 Assume that there are total number of $G$ groups, the length of $\theta$ is multiplied by $G$, and for each part, $\theta$ is parametrized in the same fashion.
 For example, for an unstructured covariance matrix, $\theta$ has $k = G * m(m+1)/2$ entries.
 
+#### Spatial covariance matrix
+
+Under spatial covariance structure, the covariance matrix is different for all subjects, if they have different visit timepoints.
+Following the notations in the previous section, for subject $i$ with total number of $m_i$ visits, we have
+
+\[
+\Sigma_{i}(m_{ij}, m_{ik}) = \sigma * f(dist(t_{m_j}, t_{m_k}))
+\]
+
+The $(m_{ij}, m_{ik})$ element of $\Sigma_{i}$ is a function of the distance between $m_{ij}$ and $m_{ik}$ visit occurring on $t_{m_{ij}}$ and $t_{m_{ik}}$.
+$t_{m_{ij}}$ is the coordinate(time) of $m_{ij}$ visit for subject $i$.
+$\sigma$ is the constant variance.
+Usually we use Euclidean distance.
+
+Currently only spatial exponential covariance structure is implemented(with only one continuous time variable).
```

Comment:
multiple coordinates supported now; at first I though it is not that necessary to support multiple coordinates because

1. the distance can be more complicated (there can be some changes to the euclidean distance, e.g, scale a specific variable, etc; but this can be done in R side)
2. extracting the terms can be complicated as there are multiple coordinates.

Anyway it is implemented now; the formula looks like  `dist(a,b)|c/d` ( I added `dist` here to avoid the confusion of adding a and b directly

Reply:
Thanks Liming, cool!
Re: formula - I think we could just say `sp_exp(a, b | c/d)` ?  Idea being everything before the `|` are the coordinates

Reply:
updated now

Reply:
awesome thanks!